ledger/recovering-indexer-integration-tests/src/test/suite/scala/com/digitalasset/platform/indexer/RecoveringIndexerIntegrationSpec.scala

```diff
@@ -348,4 +327,65 @@ object RecoveringIndexerIntegrationSpec {
 
     private class StateUpdatesFailedException extends RuntimeException("State updates failed.")
   }
+
+  class PartyOnlyQueueWriteService(
+      ledgerId: String,
+      participantId: Ref.ParticipantId,
+  )(implicit materializer: Materializer)
+      extends WritePartyService
+      with ReadService {
+
+    private val (queue, source) = Source.queue[(Offset, Update)](10).preMaterialize()
+    private val offset = new AtomicLong(0)
+    // required for the indexer to resubscribe to the update source
+    private val broadcastSource = source.runWith(BroadcastHub.sink)
+    private val stateSeenSoFar = scala.collection.mutable.Buffer.empty[(Offset, Update)]
```

Comment:
Nit:
```suggestion
    private val updatesSeenSoFar = mutable.Buffer.empty[(Offset, Update)]
```
\+ an import